src/core/annotation.js

```diff
@@ -112,6 +95,49 @@ var Annotation = (function AnnotationClosure() {
 
   Annotation.prototype = {
     /**
+     * Set the color and take care of color space conversion.
+     *
+     * @public
+     * @memberof Annotation
+     * @param {Array} color - The color array containing either 0
+     *                        (transparent), 1 (grayscale), 3 (RGB) or
+     *                        4 (CMYK) elements
+     */
+    setColor: function Annotation_setColor(color) {
+      if (!isArray(color)) {
+        return;
+      }
+
+      var rgbColor = new Uint8Array(3);
+      switch (color.length) {
+        case 0: // Transparent, which we indicate with a null value
+          this.color = null;
+          break;
+
+        case 1: // Convert grayscale to RGB
+          var grayCS = new DeviceGrayCS();
```

Comment:
Let's avoid creating lots of object by instead using `ColorSpace.singletons.gray` (similarly for rgb/cmyk below).